pkg/disruption/backend/sampler/remote.go

```diff
@@ -145,29 +149,14 @@ func StartInClusterMonitors(ctx context.Context, config *rest.Config) error {
 func deleteTestBed(ctx context.Context, kubeClient *kubernetes.Clientset) error {
 	// Remove daemonsets first to avoid trailing false-positive disruption intervals
 	dsClient := kubeClient.AppsV1().DaemonSets(namespace)
-	err := dsClient.DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{})
-	if err != nil && !apierrors.IsNotFound(err) {
+	propagationPolicy := metav1.DeletePropagationForeground
+	err := dsClient.DeleteCollection(ctx, metav1.DeleteOptions{PropagationPolicy: &propagationPolicy}, metav1.ListOptions{})
```

Comment:
deleting individual daemonsets looks like an accident. Delete the namespace.

Reply:
Its intended. We need to stop daemonset first - otherwise CRB may be removed before all pods are stopped, so some tests were showing false-positive disruption at the very end

Reply:
No longer needed after it was converted into an invariant